lib-multisrc/colamanga/src/eu/kanade/tachiyomi/multisrc/colamanga/ColaManga.kt

```diff
@@ -47,25 +63,31 @@ abstract class ColaManga(
 
     override val supportsLatest = true
 
-    private val json: Json by injectLazy()
-
-    private val intl = ColaMangaIntl(lang)
-
+    private var pages: MutableMap<String, ArrayList<Page>> = mutableMapOf()
+    private val intl = Intl(
+        Locale.getDefault().language,
+        setOf("en", "zh"),
+        lang,
+        javaClass.classLoader!!,
+    )
     private val preferences by getPreferencesLazy()
-
-    override val client = network.cloudflareClient.newBuilder()
-        .rateLimitHost(
-            baseUrl.toHttpUrl(),
-            preferences.getString(RATE_LIMIT_PREF_KEY, RATE_LIMIT_PREF_DEFAULT)!!.toInt(),
-            preferences.getString(RATE_LIMIT_PERIOD_PREF_KEY, RATE_LIMIT_PERIOD_PREF_DEFAULT)!!.toLong(),
-            TimeUnit.MILLISECONDS,
-        )
-        .addInterceptor(ColaMangaImageInterceptor())
-        .build()
-
-    override fun headersBuilder() = super.headersBuilder()
-        .add("Origin", baseUrl)
-        .add("Referer", "$baseUrl/")
+    var webView: MutableMap<String, WebView> = mutableMapOf()
```

Comment:
this will be taxing for lower end devices, when users start to open up chapter after chapter without reading completely, so there should be a upper limit to the number of cached webviews. you can just throw exceptions in `fetchImageUrl` if webview is no loner available.

use something like this to keep only 5 webviews at a time:
```kt
private val webViewCache = object : LinkedHashMap<String, WebView>() {
        override fun removeEldestEntry(eldest: MutableMap.MutableEntry<String, WebView>?): Boolean {
            val shouldRemove = size > 5
            if (shouldRemove && eldest != null) {
                eldest.value.destroy()
            }
            return shouldRemove
        }
    }
```